### DISCUSSION TITLE
Allow regex in fileFilters to use handlebar fields to limit commits

### DISCUSSION THREAD

**Discussion by JarneVdP** [renovatebot/renovate#30979] [Suggest an Idea]
### Tell us more.

Currently [fileFilters](https://docs.renovatebot.com/configuration-options/#filefilters) supports glob patterns but not regex. Based on my testing, handlebar fields are not compatible with glob pattern matching.
Ideally, I would like to use:
``` json
"fileFilters": ["{{{depName}}}", "*.gitmodules"]
```
However, this approach didn’t work for me. Hardcoding the `depName` was successful, but it’s not a viable solution since `depName` varies with each update, repository, etc. On the other hand, handlebar fields in the `postUpgradeTasks` commands do function correctly.

Currently, I’m encountering an issue where incorrect configuration of the `renovate.json` custom template results in extra files being committed per MR. If needed, I can provide more details about the error. The proposed solution would prevent this issue.

Thanks for reading and considering the idea

**Comment by rarkins** [comment]
Perhaps the true problem is that the `fileFilters` field is never _compiled_?

**Reply by JarneVdP** [reply]
Thank you for your response.
Is that something I can adjust in my renovate set up or is it a general renovate 'issue'?
The postUpgradeTasks command does convert `{{{depName}}}` to the correct string to execute the required command. For example, `git -C {{{depName}}}` -> `git -C lib/xyz`

**Reply by rarkins** [reply]
Do you have the ability to fork/edit/test locally? If so then try editing `exposedConfigOptions` to add `fileFilters`

**Reply by JarneVdP** [reply]
We forked the project and added `fileFilters` to `exposedConfigOptions` and got it running on our self-managed GitLab instance.

``` diff
diff --git a/lib/util/template/index.ts b/lib/util/template/index.ts
index f7af17fdd..88aed815b 100644
--- a/lib/util/template/index.ts
+++ b/lib/util/template/index.ts
@@ -134,6 +134,7 @@ export const exposedConfigOptions = [
   'commitMessagePrefix',
   'commitMessageSuffix',
   'commitMessageTopic',
+  'fileFilters',
   'gitAuthor',
   'group',
   'groupName',
```

Sadly we never got the handlebar in `fileFilters` to work.
```json
"postUpgradeTasks": {
	"commands": [
		"...; git -C {{{depName}}} ...; ..."
	],
	"fileFilters": ["{{{depName}}}",  "*.gitmodules"]
}
```

While the `commands` section successfully recognized and replaced the handlebars, the `fileFilters` section did not. It only worked correctly when we hardcoded the path and file, which is not a viable solution for us.

Attached is a cleaned-up debug log. Please let me know if any important parts are missing.
[debug.log.txt](https://github.com/user-attachments/files/16749187/debug.log.txt)

**Reply by rarkins** [reply]
Sorry, I don't have any time to debug it myself

**Reply by viceice** [reply]
`fileFilters` are not compiled and I don't think it's a good idea to allow it by default. it'll break globs

**Reply by JarneVdP** [reply]
I could try to debug it however I don't know how everything works under the hood nor is my typescript knowledge up to par. Is it supposed to work when adding `fileFilters` to `exposedConfigOptions`?

Is it an idea to give the option to compile `fileFilters` or is it strictly a yes/no scenario?
Alternatively, is there a different method to limit the `postUpgradeTasks` `fileFilters` dynamically?

**Reply by rarkins** [reply]
Compiling file filters will cause problems for some other use cases unfortunately
